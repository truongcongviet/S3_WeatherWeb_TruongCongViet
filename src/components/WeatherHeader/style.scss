.weatherHeader {
    display: flex;
    justify-content: space-between;
    text-align: center;
    height: 15vh;
    padding: 20px;
    font-size: 25px;
    .custom-select-cities {
        border: none;
        background: none;
        color: var(--white);
        animation: fadeIn 0.7s ease-in-out;

        .city {
            background: var(--primary-light);
            color: var(--black);
            
        }
    }
}
@keyframes fadeIn {
    0% {
      opacity: 0;
      transform: translateY(2rem);
    }
    100% {
      opacity: 1;
      transform: translateY(0);
    }
  }
