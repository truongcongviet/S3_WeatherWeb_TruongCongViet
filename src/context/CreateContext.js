import { createContext, useCallback, useContext, useReducer, Fragment } from 'react';
import { weatherReducer, initialState } from './reducers/weatherReducer'
const CreateContext = createContext({});

export const useAppContext = () => {
    const { data, setData } = useContext(CreateContext)
    const dispatch = useCallback(
        ({ type, data }) => {
            setData({ data, type })
        },
        [setData]
    )
    return [data, dispatch]
}

export const WeatherProvider = props => {
    const { children } = props;
    const [data, dispatch] = useReducer(weatherReducer, initialState);
    return (
        <CreateContext.Provider value={{ data, setData: dispatch }}>
            <Fragment>
                {children}
            </Fragment>

        </CreateContext.Provider>
    )
}

export default CreateContext;