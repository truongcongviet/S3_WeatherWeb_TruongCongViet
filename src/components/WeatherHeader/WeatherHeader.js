import './style.scss';
import { Form, Select } from 'react-bootstrap';
import { useAppContext } from '../../context/CreateContext';
import weatherServices from '../../services/weather.services';
import { GET_DATA_WEATHER } from "../../context/constant/actionGetDataWeather";
import { useState } from 'react';

const WeatherHeader = () => {
    console.log('props', useAppContext());
    const [{cities }] = useAppContext();
    const dispatch = useAppContext()[1];
    const [dataCity, setDataCity] = useState(cities)

    //get data weather from select city 
    const handleChangeCity = (e) => {
        let idCity = parseInt(e.target.value);
        weatherServices.oneCall({ lat: dataCity[idCity].lat, lon: dataCity[idCity].lon }).then(res => {
            dispatch({ type: GET_DATA_WEATHER, data: res.data });
        });
    }
    return (
        <div className="weatherHeader" role="header">
            <div className="menu-icon">
                <i className="bi bi-list h1"></i>
            </div>
            <div className="get-location">
                <div>myENV</div>
                <select className="custom-select-cities" onClick={(e) => handleChangeCity(e)}>
                    {cities.map((city, index) => {
                        return <option className="city" value={index} key={index} >{city.nameCity}</option>
                    })
                    }
                </select>
            </div>
            <div className="bell-icon">
                <i className="bi bi-bell-fill h2"></i>
            </div>
        </div>
    )
}

export default WeatherHeader;