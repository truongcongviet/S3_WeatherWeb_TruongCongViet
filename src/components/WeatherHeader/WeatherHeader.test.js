import React from 'react';
import WeatherHeader from './WeatherHeader';
import { render, screen, cleanup } from '@testing-library/react';
import { mount, configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { useAppContext, WeatherProvider } from '../../context/CreateContext'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const wrapper = mount(<WeatherProvider><WeatherHeader /> </WeatherProvider>);
describe('weatherHeader Component', () => {
    test("expect to render WeatherHeader component", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const WeatherHeaderComponent = wrapper.find(".weatherHeader");
        expect(WeatherHeaderComponent.length).toBe(1);;
    });
    
})
