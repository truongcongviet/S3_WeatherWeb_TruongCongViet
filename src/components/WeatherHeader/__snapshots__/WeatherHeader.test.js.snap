// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weatherHeader Component expect to render WeatherHeader component 1`] = `
<WeatherProvider>
  <WeatherHeader>
    <div
      className="weatherHeader"
      role="header"
    >
      <div
        className="menu-icon"
      >
        <i
          className="bi bi-list h1"
        />
      </div>
      <div
        className="get-location"
      >
        <div>
          myENV
        </div>
        <select
          className="custom-select-cities"
          onClick={[Function]}
        >
          <option
            className="city"
            key="0"
            value={0}
          >
            Hồ Chí Minh
          </option>
          <option
            className="city"
            key="1"
            value={1}
          >
            Singapore
          </option>
          <option
            className="city"
            key="2"
            value={2}
          >
            Jerusalem
          </option>
          <option
            className="city"
            key="3"
            value={3}
          >
            California
          </option>
          <option
            className="city"
            key="4"
            value={4}
          >
            AlasKa
          </option>
        </select>
      </div>
      <div
        className="bell-icon"
      >
        <i
          className="bi bi-bell-fill h2"
        />
      </div>
    </div>
  </WeatherHeader>
   
</WeatherProvider>
`;
